src/main/java/assignment2/services/DiskStorage.java

```diff
@@ -0,0 +1,78 @@
+package assignment2.services;
+
+import assignment2.Application;
+import assignment2.models.User;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.ArrayList;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DiskStorage {
+
+  final static Logger logger = LoggerFactory.getLogger(Application.class);
+
+  /**
+   * Function to save the users to disk
+   *
+   * @param users ArrayList of User objects
+   */
+  public static void saveToDisk(ArrayList<User> users) {
+    try {
+      FileOutputStream fos = new FileOutputStream("userdetails.ser");
```

Comment:
on each thread, you are creating object of stream. What is the need of creating instance everytime?